src/OpenSage.Launcher/Program.cs

```diff
@@ -60,9 +63,67 @@ public static void Main(string[] args)
 
         public static void Run(Options opts)
         {
-            logger.Info("Starting...");
+            logger.Info("Starting...");
+
+            var DetectedGame = opts.Game;
+            var GameFolder = opts.GamePath;
+            var ForceUseEnvironmentGamePath = true;
 
-            var definition = GameDefinition.FromGame(opts.Game);
+            if (GameFolder == null)
+            {
+                GameFolder = Environment.CurrentDirectory;
+            }
+            if (File.Exists(Path.Combine(GameFolder, "generals.exe")) && File.Exists(Path.Combine(GameFolder, "INI.big")))
```

Comment:
I think it would make sense to put this in `IGameDefinition`, so each game definition has a **Probe** method returning a bool wether the path is the correct game installation folder. Then you could just iterate over all GameDefinition and remove this logic here

Reply:
done